### PR TITLE
Lookup command

### PR DESCRIPTION
**Pull request by ro0gr**
#patch#

Partially addresses #92 (only `new-baseline` and `build-report` tasks) 

 - [x] test on Windows

### CODE REVIEW

**Comment by ember-cli-visual-acceptance** [comment]
# Visual Acceptance tests:
### No changes

**Comment by ro0gr** [comment]
https://travis-ci.org/ciena-blueplanet/ember-cli-visual-acceptance/builds/158293325#L1703

Here you can see doubled watchman message. 
It happens due to  hard-coded watcher detection in the [`Command.validateAndRun()` method](https://github.com/ember-cli/ember-cli/blob/2bce175ce188ae0f47a55fa23d6affe2d768ddfb/lib/models/command.js#L219).

It's unpleasant UX regression but I believe it's not a blocker for enabling windows support.

**Comment by EWhite613** [comment]
Thanks for this @ro0gr 

I believe we always had that double watchman message. https://travis-ci.org/ciena-blueplanet/ember-cli-visual-acceptance/builds/157331274#L1696-L1707 (old build)

But this looks even better since the output isn't going through the previous spawn stdout.on

File: lib/utils.js [hunk 1]
```diff
@@ -35,7 +36,7 @@ function mkdirpSync (dirpath) {
   }
 }
 
-function runCommand (command, args, ignoreStdError) {
+function runExecutable (command, args, ignoreStdError) {
   return new RSVP.Promise(function (resolve, reject) {
     var child = spawn(command, args, {
```
Comment (EWhite613) [inline lib/utils.js]: @ro0gr Does this issue https://github.com/ciena-blueplanet/ember-cli-visual-acceptance/issues/92 still occur here when executing `git`? Even though I doubt anyone would run a CI build on Windows.

Comment (ro0gr) [inline lib/utils.js]: @EWhite613 I haven't touch calls to `git` in this pr. So I think the issue still persist for `teamcity-bitbucket-visual-acceptance` and `travis-visual-acceptance` commands.

Anyway the goal of this pr is to enable running of `ember` commands via `lookup-command`. I believe we should fix `git` in a separate pr.

Comment (ro0gr) [inline lib/utils.js]: I've updated pr description a bit to match the goal of this pr.

**Comment by ro0gr** [comment]
I just had a chance to test `new-baseine` and `build-report` commands against windows. It just works.

**Comment by ro0gr** [comment]
I've just tested `runExecutable` and looks like I was wrong about `spawn`.
```javascript
var runExecutable = require('ember-cli-visual-acceptance/lib/utils').runExecutable
runExecutable('git', ['help'])
```
Outputs:
```powershell
b$rsvp$promise$$Promise {
_id: 14,
_label: undefined,
_state: undefined,
_result: undefined,
_subscribers: [] }
usage: git [--version] [--help] [-C <path>] [-c name=value]
         [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
         [-p | --paginate | --no-pager] [--no-replace-objects] [--bare]
         [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
         <command> [<args>]
         ...
```
So the `spawn` is able to resolve executables from the PATH. The reason it doesn't currently work for node executables is an absence of `.cmd` extension:
```javascript
 runExecutable('ember.cmd', ['help', 'br'])
```
Outputs:
```powershell
lib$rsvp$promise$$Promise {
  _id: 15,
  _label: undefined,
  _state: undefined,
  _result: undefined,
  _subscribers: [] }
> Requested ember-cli commands:



ember build-report <options...>
  Create report
  aliases: br
  --report-directory (String) (Default: visual-acceptance-report) Create Report off visual acceptance tests
```

So basically it means that this pr should enable windows support. I hope

**Comment by EWhite613** [comment]
@ro0gr do your images get saved? I tried `ember new-baseline` quickly and it doesn't seem to save the images

**Comment by EWhite613** [comment]
I seem to be getting 404'd on POST `/passed`. Which is odd since all the others go through fine

**Comment by ro0gr** [comment]
Hm.. that's interesting. Do you mean windows? Unfortunately I don't have access to windows pc at this moment. Can check it tomorrow.

**Comment by EWhite613** [comment]
@ro0gr I added some path fixes to your PR. I meant to make a PR to ur branch but I guess I somehow pushed directly to ur branch

**Comment by EWhite613** [comment]
@ro0gr ya I meant windows. Shoulda clarified. Let me know if my commits look good to u and we can merge this in

**Comment by ro0gr** [comment]
Yes, your path fixes seem reasonable to me

**Comment by ro0gr** [comment]
> can run IE11/Edge in a VM, but a native CI machine would be :+1:
I think it would be the best if we enable windows CI before merge this pr. @EWhite613 Is it possible in a short term or will do it later?

**Pull request merged**